src/validators.ts

```diff
@@ -114,3 +114,21 @@ export function optional<T>(
     return validator(val);
   };
 }
+
+export function optionalArray<T>(validator: Validator<T>): Validator<T | T[]>;
+export function optionalArray<T>(
+  validator: Validator<T>,
+  result: T | T[],
+): Validator<T | T[]>;
+export function optionalArray<T>(
+  validator: Validator<T>,
+  result?: T | T[],
+): Validator<T | T[] | undefined> {
+  return (val: any): T | T[] => {
+    if (Array.isArray(val)) {
+      return result ?? val;
```

Comment:
I'm a little confused by this, if the received value is an array this defaults to returning a passed default result?

I think we just don't need the default result at all and instead want to run the validator over every element in the array if `val` is an array.